rspace/src/main/scala/coop/rchain/rspace/RSpaceOps.scala

```diff
@@ -79,28 +76,27 @@ abstract class RSpaceOps[F[_]: Async: Log: Metrics: Span, C, P, A, K](
   protected[this] val produceCommLabel     = "comm.produce"
   protected[this] val produceTimeCommLabel = "comm.produce-time"
 
-  protected val historyRepositoryAtom: AtomicAny[HistoryRepository[F, C, P, A, K]] = AtomicAny(
-    historyRepository
-  )
+  protected val historyRepositoryRef: F[Ref[F, HistoryRepository[F, C, P, A, K]]] =
+    Ref.of[F, HistoryRepository[F, C, P, A, K]](historyRepository)
```

Comment:
This is the place. Previously historyRepositoryAtom has been created when RSpaceOps are created, but now with your change it is created next to the place when the content of the Ref is read.